examples/common/pigweed/rpc_services/Descriptor.h

```diff
@@ -0,0 +1,89 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "descriptor_service/descriptor_service.rpc.pb.h"
+
+#include <app-common/zap-generated/af-structs.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+
+namespace chip {
+namespace rpc {
+
+class Descriptor : public pw_rpc::nanopb::Descriptor::Service<Descriptor>
+{
+public:
+    virtual ~Descriptor() = default;
+
+    virtual void DeviceTypeList(const ::chip_rpc_Endpoint & request, ServerWriter<::chip_rpc_DeviceType> & writer)
+    {
+        uint16_t index = emberAfIndexFromEndpoint(request.endpoint);
+        chip_rpc_DeviceType out{ .device_type = emberAfDeviceIdFromIndex(index) };
```

Comment:
Do we care that if the endpoint passed in was invalid this will do some random out-of-bounds read?

Reply:
Good catch, fixed. 
Should this also be fixed in the cluster? https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/descriptor/descriptor.cpp#L92

Reply:
In the cluster, the code can't be reached for an invalid endpoint: there are checks higher up the callstack that the entire path being passed in corresponds to an enabled endpoint, with errors if it does not.